header.php

```diff
@@ -34,7 +34,7 @@
 					<?php twentynineteen_post_thumbnail(); ?>
 					<?php the_post(); ?>
 					<?php $discussion = ! is_page() && twentynineteen_can_show_post_thumbnail() ? twentynineteen_get_discussion_data() : null; ?>
-					<div class="<?php echo ( ! empty( $discussion ) && count( $discussion->responses ) > 0 ) ? 'entry-header has-discussion' : 'entry-header'; ?>">
+					<div class="<?php echo ( ! empty( $discussion ) && count( $discussion->authors ) > 0 ) ? 'entry-header has-discussion' : 'entry-header'; ?>">
```

Comment:
If the intention is to count the number of responses, why change it to counting the number of authors instead? Just dumping `count()` did the trick for me locally. Since `$discussion->responses` returns a string value (which probably should be reviewed), wrapping it in `absint()` ensures we get an integer:

```html
<div class="<?php echo ( ! empty( $discussion ) && absint( $discussion->responses ) > 0 ) ? 'entry-header has-discussion' : 'entry-header'; ?>">
```

Reply:
I made your suggested change. Since you asked, the author count seemed more reliable at the time because `$discussion->responses` is derived from `get_comment_count()` and its return value can be filtered. 

What can happen is another plugin, like muut, will filter the count to be zero (or even a placeholder to be filled later with JS). So there's a chance a post with comments will have a filtered value be zero, but that's probably an edge case. Worse case scenario is the CSS class won't be added there.